packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -599,7 +599,7 @@ export const RemoteServerFactory = {
             // do that now so that the rest of the code don't have to deal
             // with these headers, which can be large and may be accidentally
             // forwarded to other servers.
-            X_HEADERS_TO_REMOVE.forEach((key) => {
+            X_HEADERS_TO_REMOVE_ORIGIN.forEach((key) => {
```

Comment:
Non-blocking longer term note: it might be useful to move this header removal to platform middleware to fully isolate the values from customer code